content/posts/2021/reana-client-0.7.3.md

```diff
@@ -0,0 +1,346 @@
+
+```console
+$ reana-client validate -f reana.yaml
+==> Verifying REANA specification file... my-analysis/reana.yaml
+  -> SUCCESS: Valid REANA specification file.
+==> Verifying workflow parameters and commands...
+  -> WARNING: REANA input parameter "sleeptime" does not seem to be used.
+```
+
+The specification remains valid, as it is well-formed, however, a
+warning is displayed indicating that `sleeptime` parameter is
+defined but it does not seem to be used in the workflow
+specification.
+
+Let us fix this problem by adding this unsused parameter to the
+command, but imagine that we make a typo when adding it and we write
+`sleptime` instead of `sleeptime`:
+
+```diff
+       - name: run-script
+         environment: 'python:3.9-slim'
+         commands:
+-          - python "${script_path}"
++          - python "${script_path}" "${sleptime}"
+```
+
+Let us call the validator to see the output:
+
+```console
+$ reana-client validate -f reana.yaml
+==> Verifying REANA specification file... my-analysis/reana.yaml
+  -> SUCCESS: Valid REANA specification file.
+==> Verifying workflow parameters and commands...
+  -> WARNING: REANA input parameter "sleeptime" does not seem to be used.
+  -> WARNING: Serial parameter "sleptime" found on step "run-script" is not defined in input parameters.
+```
+
+As we made a typo using the parameter, `sleptime` counts as a new one.
+It is present in the workflow commands of step `run-script` but is
+not defined. Let us fix the typo and call the validator again:
+
+```diff
+     - code/myanalysis.py
+   parameters:
+     script_path: code/myanalysis.py
++    sleeptime: 10
+ workflow:
+   type: serial
+   specification:
+     steps:
+       - name: run-script
+         environment: 'python:3.9-slim'
+         commands:
+-          - python "${script_path}" "${sleptime}"
++          - python "${script_path}" "${sleeptime}"
+```
+
+```console
+$ reana-client validate -f reana.yaml
+==> Verifying REANA specification file... my-analysis/reana.yaml
+  -> SUCCESS: Valid REANA specification file.
+==> Verifying workflow parameters and commands...
+```
+
+Now everything is settled, the warnings disappear as the new
+parameter is defined and used properly.
+
+This validation is also compatible with Yadage and CWL workflows,
+where more advanced verifications are carried out as parameters
+can be defined at multiple levels.
+
+### Workflow dangerous operations verification
+
+Over time, we have identified certain commands or operations that
+can potentially cause problems when the workflows are run in the
+REANA Cluster. It is our mission to detect these kinds of operations
+and warn users that their execution might cause errors.
+
+Some of these operations are running commands as the system superuser
+(`sudo`) or changing the working directory where the workflow is run
+instead of using the workflow workspace.
+
+Let us modify our previous example to execute the Python script as
```

Comment:
Let us modify our previous example to illustrate how this works. Imagine we would like to execute the command as superuser:

(and drop the "which we strongly..." part that was addressed above)